src/vs/editor/browser/controller/mouseHandler.ts

```diff
@@ -424,6 +450,36 @@ class MouseDownOperation extends Disposable {
 		// Overwrite the detail of the MouseEvent, as it will be sent out in an event and contributions might rely on it.
 		e.detail = this._mouseState.count;
 
+		if (!this._mouseState.altKey // we don't support multiple mouse
+			&& e.detail < 2 // only single click on a selection can work
+			&& !this._isActive // the mouse is not down yet
+			&& !this._currentSelection.isEmpty() // we don't drag single cursor
+			&& this._currentSelection.containsPosition(position.position) // single click on a selection
+		) {
+			this._isActive = true;
+			this._dropTarget = this._createMouseTarget(e, true);
+
+			this._mouseMoveMonitor.startMonitoring(
+				createMouseMoveEventMerger(null),
+				this._mouseDragThenMoveEventHandler.handler,
+				() => {
+					if (this._mouseState.lastMouseDownEvent !== 'drag') {
```

Comment:
when can this happen?

IMHO the mouseState is set to `drag` or `mouseDown` and it should not change while the mouse is pressed (i.e. while the global mouse monitoring is active).